### PR TITLE
Check app init concurrency

### PR DESCRIPTION
**Pull request by konstantin-msft**

### CODE REVIEW

File: lib/msal-browser/test/app/PublicClientApplication.spec.ts [hunk 1]
```diff
@@ -136,6 +136,75 @@ describe("PublicClientApplication.ts Class Unit Tests", () => {
     });
 
     describe("intialize tests", () => {
+        jest.setTimeout(9999999);
```
Comment (tnorling) [inline lib/msal-browser/test/app/PublicClientApplication.spec.ts]: This is a pretty high timeout, anything smaller we can use?

File: lib/msal-browser/test/app/PublicClientApplication.spec.ts [hunk 2]
```diff
@@ -136,6 +136,75 @@ describe("PublicClientApplication.ts Class Unit Tests", () => {
...
+        it("serializes concurrent calls", async () => {
+            const config = {
+                auth: {
+                    clientId: TEST_CONFIG.MSAL_CLIENT_ID
+                },
+                system: {
+                    allowNativeBroker: true
+                }
+            };
+            const samples = 100;
```
Comment (tnorling) [inline lib/msal-browser/test/app/PublicClientApplication.spec.ts]: Why do we need to run this test 100 times?

File: lib/msal-browser/test/app/PublicClientApplication.spec.ts [hunk 3]
```diff
@@ -136,6 +136,75 @@ describe("PublicClientApplication.ts Class Unit Tests", () => {
...
+                const handshakeSpy: sinon.SinonSpy = sinon.spy(NativeMessageHandler.prototype, "sendHandshakeRequest");
+                const initEventSpy = sinon.stub(EventHandler.prototype, "emitEvent");
+                let mcPort: MessagePort;
+
+                const eventHandler = function (event: MessageEvent) {
+                    event.stopImmediatePropagation();
+                    const request = event.data;
+                    const req  = {
+                        channel: "53ee284d-920a-4b59-9d30-a60315b26836",
+                        extensionId: "test-ext-id",
```
Comment (tnorling) [inline lib/msal-browser/test/app/PublicClientApplication.spec.ts]: What about cases where the first handshake (to the specific extensionId) fails and the second handshake (without extensionId) is executed?

File: lib/msal-browser/test/app/PublicClientApplication.spec.ts (hunk 1 above)
Comment (konstantin-msft) [reply lib/msal-browser/test/app/PublicClientApplication.spec.ts]: It's just a one-time test to verify that `app.init` calls are serialized by the event loop. I have no plans to merge it.

File: lib/msal-browser/test/app/PublicClientApplication.spec.ts (hunk 3 above)
Comment (tnorling) [reply lib/msal-browser/test/app/PublicClientApplication.spec.ts]: Similarly, what's the behavior when one or more handshakes fail?

File: lib/msal-browser/test/app/PublicClientApplication.spec.ts (hunk 2 above)
Comment (konstantin-msft) [reply lib/msal-browser/test/app/PublicClientApplication.spec.ts]: To make sure the serialization pattern is consistent.

File: lib/msal-browser/test/app/PublicClientApplication.spec.ts [hunk 4]
```diff
@@ -136,6 +136,75 @@ describe("PublicClientApplication.ts Class Unit Tests", () => {
...
+                        }
+                    };
+
+                    mcPort = postMessageSpy.args[0][2][0];
+                    mcPort.postMessage(req);
+                };
+                window.addEventListener("message", eventHandler, true);
+
+                const promises = [];
+                for (let x = 1; x <= concurrency; x++) {
```
Comment (sameerag) [inline lib/msal-browser/test/app/PublicClientApplication.spec.ts]: nit: index to `0`?

**Review by fepimentel — APPROVED** [review]

**Pull request closed without merging**